presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSubsetNodeSetSupplier.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.execution.QueryState;
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.metadata.InternalNodeManager;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.Node;
+import com.facebook.presto.spi.QueryId;
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSetMultimap;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.execution.scheduler.NodeSelectionHashStrategy.CONSISTENT_HASHING;
+import static com.facebook.presto.metadata.InternalNode.NodeStatus.ALIVE;
+import static com.facebook.presto.spi.NodeState.ACTIVE;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.lang.Math.ceil;
+import static java.lang.String.format;
+
+/**
+ * This WIP class is responsible for assigning a subset of the cluster nodes
+ * to a requesting query.
+ * The size of the subset is currently static, but it can by defined by configuration
+ * or can be driven by fairness algorithm depending on query metadata.
+ */
+public class FixedSubsetNodeSetSupplier
+        implements NodeSetSupplier, Closeable
```

Comment:
Maybe its too soon: But should we share some code with the DefaultNodeSupplier ? 

I am also very unclear in the difference b/w a NodeSet and the older NodeMap, can you clarify that in comments somewhere ?

Reply:
Yup. Current code is to quicly test the feasibility and wins. Will def look into code restructuring and cleanup. 

> NodeSet and the older NodeMap,

No difference. Will undo this renaming and retain NodeMap